app/components/primer/alpha/segmented_control.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    # Add a general description of component here
+    # Add additional usage considerations or best practices that may aid the user to use the component correctly.
+    # @accessibility Add any accessibility considerations
+    class SegmentedControl < Primer::Component
+      status :alpha
+
+      ICON_ONLY_DEFAULT = false
+      ICON_ONLY_OPTIONS = [ICON_ONLY_DEFAULT, true, :whenNarrow].freeze
```

Comment:
This is a bit of a strange compromise that I did until we make updates to our responsive guidelines for `when_narrow: { icon_only: true }` I'm willing to change how this is if someone has concerns.

Reply:
I don't think `icon_only` should be on the parent - I'd recommend either:
- A different child component (this is what we do in React: https://primer.style/react/SegmentedControl#with-icons-only)
- A prop on the child Button component

I recommend doing a different child component because icon-only buttons will show a tooltip, so the functionality is a little different.

Reply:
Are we able to support both responsive variants in PVC?
1. icon-only
2. swap with an Dropdown

Reply:
This got buried in "hidden items". Mentioning @jonrohan so it doesn't get lost.

Reply:
It looks like the feedback about the `icon_only` arg moving from the parent to the children has been addressed 👍